packages/aws-cdk-lib/aws-stepfunctions/lib/state-machine.ts

```diff
@@ -452,6 +472,33 @@ export class StateMachine extends StateMachineBase {
       }
     }
 
+    if (props?.kmsDataKeyReusePeriodSeconds && !isValidKmsDataKeyReusePeriodSeconds(props.kmsDataKeyReusePeriodSeconds)) {
+      throw new Error('kmsDataKeyReusePeriodSeconds needs to be a value between 60 and 900');
+    }
+
+    if (props?.kmsKey) {
+      props?.kmsKey.addToResourcePolicy(new iam.PolicyStatement({
```

Comment:
Optional chaining is not necessary here and the other places you added. This line will only execute when the `kmsKey` is provided through `props` and there the code at the start of the constructor already tries to access properties of `props`.

Reply:
Ah I see, will update that, thanks